### PR TITLE
do not error if sha cannot be found on github

### PR DESCRIPTION
**Pull request by kellyoung**
https://github.com/mapbox/stork/issues/27

@vsmart and I are working on the issue above where the function errors out when a gitsha is not found.

We are working on seeing if the `gitsha` exists with a call to the github api, and if it does not exist we do not have to return an error and can just exit the function. If the `gitsha` does exist, then we know it is an error we would want to look into.

### CODE REVIEW

**Comment by kellyoung** [comment]
👋 @vsmart 
I fixed the tests to pass. Finally figured out what the config was to be passed in with the `get` request. I'll leave comments on the code to explain what I changed since we finished our pair programming session.

I think there's still some work that can be done. There's some repetitive parts of the code, so we could spend some time cleaning up and organizing the code. 🤗

File: lambda.js [hunk 1]
```diff
@@ -499,12 +508,45 @@ stork.status = (event, context, callback) => {
...
 
+        console.log(uri);
+        console.log(config);
         return got.post(uri, config);
       })
       .then(() => callback())
       .catch((err) => {
-        console.log(err);
-        callback(err);
+        const shaToken = jwt.sign(
```
Comment (kellyoung) [inline lambda.js]: 👋 @vsmart 
I got this from https://github.com/mapbox/stork/blob/shaerror/lambda.js#L32-L49
This goes into `got.get` as the second argument.

File: test/lambda.test.js [hunk 2]
```diff
@@ -1260,6 +1260,182 @@ test('[lambda] status: success', (assert) => {
...
+    .onCall(1).callsFake(() => Promise.resolve({
+      statusCode: 200,
+      statusMessage: 'OK',
+      body: {
+        sha: '12345shaexists'
+      }
+    }));
+
+  sinon.stub(got, 'post')
+    .onCall(0).callsFake(() => Promise.resolve({
```
Comment (kellyoung) [inline test/lambda.test.js:1465]: @vsmart I just added the `post` call that results in the token, similar to what we did for the `gitsha not found` tests.

File: test/lambda.test.js [hunk 3]
```diff
@@ -1260,6 +1260,182 @@ test('[lambda] status: success', (assert) => {
...
+          sourceVersion: '12345shaexists',
+          source: { location: 'https://github.com/mapbox/stork' }
+        }
+      ]
+    }));
+  });
+
+  sinon.stub(got, 'get')
+    .onCall(0).callsFake(() => Promise.resolve())
+    .onCall(1).callsFake(() => Promise.resolve({
```
Comment (kellyoung) [inline test/lambda.test.js:1456]: This was `reject`, so I changed it to `resolve` as it's not an error.

File: test/lambda.test.js [hunk 4]
```diff
@@ -1260,6 +1260,182 @@ test('[lambda] status: success', (assert) => {
...
+      statusCode: 422,
+      statusMessage: 'Unprocessable Entity'
+    }));
+
+  lambda.status(fakeStatusEvent, {}, (err) => {
+    const args = JSON.parse(JSON.stringify(got.post.args[0]));
+    const auth = args[1].headers.Authorization.replace('Bearer ', '');
+    const decoded = jwt.verify(auth, publicKey, { algorithms: ['RS256'] });
+
+    assert.equal(err.statusMessage, 'Unprocessable Entity', 'errors');
```
Comment (kellyoung) [inline test/lambda.test.js:1478]: I changed this from `err.message` to `err.statusMessage`, as we are stubbing what the error throws and the error we threw has a property of `statusMessage` and not `message`

File: lambda.js (hunk 1 above)
Comment (vsmart) [reply lambda.js]: Ah. Nice catch! 👍  I wonder if we could outsource this logic into a generateShaToken function that we can reuse.

**Comment by vsmart** [comment]
👋 Hi @kellyoung, this looks great - thanks so much for fixing these tests. I agree, we can take a final session to clean up a bit and then merge + deploy \o/

File: lambda.js (hunk 1 above)
Comment (rclark) [reply lambda.js]: Why do you need to create another github token? Why can't you continue to use the one that was generated in the initial round of requests?

**Comment by vsmart** [comment]
👍 This looks good to go, let's stage it, then merge & deploy! 🚢

**Comment by emilymcafee** [comment]
@kellyoung @vsmart really nice work on this. Thanks for picking it up.

bitmoji

**Pull request merged**